testing/test_session.py

```diff
@@ -253,3 +253,28 @@ def pytest_sessionfinish():
     """)
     res = testdir.runpytest("--collect-only")
     assert res.ret == EXIT_NOTESTSCOLLECTED
+
+
+def test_rootdir_option_arg(testdir):
+    rootdir = testdir.mkdir("root")
+    rootdir.mkdir("tests")
+    testdir.makepyfile("""
+        import os
+        def test_one():
+            assert os.path.isdir('.cache')
+    """)
+
+    result = testdir.runpytest()
+    result.stdout.fnmatch_lines(["*AssertionError*"])
+
+    result = testdir.runpytest("--rootdir=root")
+    result.stdout.fnmatch_lines(["*1 passed*"])
+
+
+def test_rootdir_wrong_option_arg(testdir):
+    rootdir = testdir.mkdir("root")
+    testsdir = rootdir.mkdir("tests")
+    testsdir.join("test_one.py").write("def test_one():\n    assert 1")
```

Comment:
This can be written as:

```python
testdir.makepyfile("""
    def test_one():
        assert 1
""")
``` 

Which is more readable IMO. This call will create a file `test_rootdir_wrong_option_arg.py` in the testdir directory (same name as the test), which is not really relevant for his this test.

Actually I don't even think the file or the directories is relevant for the test at all, so we can remove this preparation from this test.

Reply:
It seems that I can't create file in root/tests folder by this code. It will created in the testdir folder. But tests runs fine, so, fixed =)